src/views/ShareCredentialSelect/ShareCredentialSelect.tsx

```diff
@@ -100,15 +84,36 @@ export function ShareCredentialSelect({
 
   const identities = useIdentities().data;
 
-  const hasSome = useBooleanState();
+  const credentials = useCredentials();
+
+  const data = usePopupData<ShareInput>();
+  const { credentialRequest } = data;
+  const { cTypes } = credentialRequest;
+  const cTypeHashes = cTypes.map(({ cTypeHash }) => cTypeHash);
+
+  const matchingCredentials = credentials?.filter((credential) =>
+    cTypeHashes.includes(credential.request.claim.cTypeHash),
+  );
 
   const ref = useRef<HTMLElement>(null);
 
-  if (!identities) {
+  if (!identities || !matchingCredentials) {
     return null; // storage data pending
   }
 
-  const identitiesList = sortBy(Object.values(identities), 'index');
+  const noMatchingCredentials = matchingCredentials.length === 0;
+
+  const matchingCredentialDids = matchingCredentials.map(
+    (credential) => credential.request.claim.owner,
+  );
+  const identitiesWithMatchingCredentials = Object.values(identities).filter(
+    (identity) => matchingCredentialDids.includes(identity.did),
+  );
+
+  const sortedIdentities = sortBy(
+    Object.values(identitiesWithMatchingCredentials),
```

Comment:
Why `Object.values` here? `identitiesWithMatchingCredentials` seems to be an array.